src/internal/io.js

```diff
@@ -145,17 +145,19 @@ export function flush(channel) {
   return effect(FLUSH, channel)
 }
 
+const getAsEffectFn = type => effect => effect && effect[IO] && effect[type]
```

Comment:
Its not really a `get` but more of a `create` function here. I think `createAsEffectType` is a sufficient name here.